usr.sbin/rtadvd/config.c

```diff
@@ -912,6 +912,58 @@ getconfig(struct ifinfo *ifi)
 		}
 		free(dns);
 	}
+
+	/*
+	 * handle pref64
+	 */
+	rai->rai_pref64.p64_enabled = false;
+
+	if ((addr = (char *)agetstr("pref64", &bp))) {
+		if (inet_pton(AF_INET6, addr, &rai->rai_pref64.p64_prefix) != 1) {
+			syslog(LOG_ERR, "<%s> inet_pton failed for %s",
+			    __func__, addr);
+		} else {
+			rai->rai_pref64.p64_enabled = true;
+
+			switch (val64 = agetnum("pref64len")) {
+			case -1:
+			case 96:
+				rai->rai_pref64.p64_plc = 0;
+				break;
+			case 64:
+				rai->rai_pref64.p64_plc = 1;
+				break;
+			case 56:
+				rai->rai_pref64.p64_plc = 2;
+				break;
+			case 48:
+				rai->rai_pref64.p64_plc = 3;
+				break;
+			case 40:
+				rai->rai_pref64.p64_plc = 4;
+				break;
+			case 32:
+				rai->rai_pref64.p64_plc = 5;
+				break;
+			default:
+				syslog(LOG_ERR, "prefix length %" PRIi64
+				       "on %s is invalid; disabling PREF64",
+				       val64, ifi->ifi_ifname);
+				rai->rai_pref64.p64_enabled = 0;
+				break;
+			}
+
+			/* this logic is from RFC8781 section 4.1 */
```

Comment:
Very minor nit: I feel like "RFC 8781" would be more consistent here, rather than "RFC8781". I think this a great comment, though.

Reply:
And "This" and a "." at the end if you are touching it for whitespace already.

Reply:
both fixed.